lib/hyrax/transactions/steps/add_file_sets.rb

```diff
@@ -25,6 +25,14 @@ def initialize(handler: Hyrax::WorkUploadsHandler)
         # @return [Dry::Monads::Result]
         def call(obj, uploaded_files: [], file_set_params: [])
           if @handler.new(work: obj).add(files: uploaded_files, file_set_params: file_set_params).attach
+            if obj.lease
```

Comment:
a new file set gets added to the work in this `#call` method, which happens _after_ "steps/save.rb". doing this call here correctly puts the lease on new files and existing ones.